src/test/kotlin/hwr/oop/projects/peakpoker/core/game/GameTest.kt

```diff
@@ -96,6 +96,203 @@ class GameTest : AnnotationSpec() {
+        val exception = shouldThrow<IllegalArgumentException> {
+            Game(
+                1021, 20, 10,
+                listOf(Player("Hans"), Player("Peter"), Player("Max"))
+            )
+        }
+
+        assertThat(exception.message).isEqualTo("Big blind amount must be greater than or equal to small blind amount")
+    }
+
+    @Test
+    fun `calculatePot returns sum of all player bets`() {
+        val player1 = Player("Hans")
+        val player2 = Player("Peter")
+        val player3 = Player("Max")
+        val testGame = Game(1031, 10, 20, listOf(player1, player2, player3))
+
+        player1.raiseBet(10)
+        player2.raiseBet(30)
+        player3.raiseBet(20)
+
+        assertThat(testGame.calculatePot()).isEqualTo(60)
+    }
+
+    @Test
+    fun `makeTurn advances currentPlayerIndex to next player`() {
+        val testGame = Game(
+            1032, 10, 20,
+            listOf(Player("Hans"), Player("Peter"), Player("Max"))
+        )
+
+        val initialPlayer = testGame.getCurrentPlayer()
+        testGame.makeTurn()
+        val nextPlayer = testGame.getCurrentPlayer()
+
+        assertThat(nextPlayer.name).isNotEqualTo(initialPlayer.name)
+        assertThat(testGame.currentPlayerIndex).isEqualTo(0)
+    }
+
+    @Test
+    fun `checkPlayerValidity returns false for existing players and true for new players`() {
+        val testGame = Game(
+            1033, 10, 20,
+            listOf(Player("Hans"), Player("Peter"), Player("Max"))
+        )
+
+        val existingPlayer = Player("Hans")
+        val newPlayer = Player("Sara")
+
+        assertThat(testGame.checkPlayerValidity(existingPlayer)).isFalse()
+        assertThat(testGame.checkPlayerValidity(newPlayer)).isTrue()
+    }
+
+    @Test
+    fun `getHighestBet handles edge cases with equal bets`() {
+        val player1 = Player("Hans")
+        val player2 = Player("Peter")
+        val player3 = Player("Max")
+        val testGame = Game(1040, 10, 20, listOf(player1, player2, player3))
+
+        player1.raiseBet(30)
+        player2.raiseBet(30)
+        player3.raiseBet(30)
+
+        assertThat(testGame.getHighestBet()).isEqualTo(30)
+    }
+
+    @Test
+    fun `getHighestBet works with zero bets`() {
+        val player1 = Player("Hans")
+        val player2 = Player("Peter")
+        val player3 = Player("Max")
+        val testGame = Game(1041, 10, 20, listOf(player1, player2, player3))
+
+        assertThat(testGame.getHighestBet()).isEqualTo(0)
+    }
+
+    @Test
+    fun `getHighestBet returns correct value when some bets are the same`() {
+        val player1 = Player("Hans")
+        val player2 = Player("Peter")
+        val player3 = Player("Max")
+        val testGame = Game(1042, 10, 20, listOf(player1, player2, player3))
+
+        player1.raiseBet(10)
+        player2.raiseBet(30)
+        player3.raiseBet(10)
+
+        assertThat(testGame.getHighestBet()).isEqualTo(30)
```

Comment:
Same as above:
```suggestion
        assertThat(testGame.getHighestBet()).isEqualTo(50)
```
(10+30+10=50)